crates/ln-dlc-node/src/ln/event_handler.rs

```diff
@@ -434,14 +441,42 @@ where
             Event::ChannelClosed {
                 channel_id,
                 reason,
-                user_channel_id: _,
+                user_channel_id,
             } => {
-                let channel = hex::encode(channel_id);
+                let channel_id = hex::encode(channel_id);
+                let user_channel_id = Uuid::from_u128(user_channel_id).to_string();
                 tracing::info!(
-                    %channel,
+                    %user_channel_id,
+                    %channel_id,
                     ?reason,
                     "\nChannel closed",
                 );
+
+                if let Some(mut channel) = self.storage.get_channel(&user_channel_id)? {
+                    if !vec![ChannelState::Pending, ChannelState::Open]
+                        .contains(&channel.channel_state)
+                    {
+                        tracing::warn!(%channel_id, %user_channel_id, "Unexpected state transition. Expected channel state to be either 'Pending' or 'Open', but was '{:?}'", channel.channel_state);
+                    }
+
+                    match reason {
+                        ClosureReason::CounterpartyForceClosed { .. }
+                        | ClosureReason::CommitmentTxConfirmed => {
+                            channel.channel_state = ChannelState::ForceClosedRemote;
+                        }
+                        ClosureReason::HolderForceClosed { .. } => {
+                            channel.channel_state = ChannelState::ForceClosedLocal;
+                        }
+                        _ => {
+                            // TODO: we can't distinguish here whether we or the counter party
+                            // initiated the close of the channel.
```

Comment:
🤔 I don't think we need a TODO here. Some of those `ClosureReason`s are not initiated by either party anyway. I would just document what each variant means in `ChannelState`.

Reply:
removed the todo